polkadot/node/core/prospective-parachains/src/fragment_tree.rs

```diff
@@ -869,6 +869,9 @@ impl FragmentTree {
 
 			// Short-circuit the search if we've found the right length. Otherwise, we'll
 			// search for a max.
+			// Taking the first best selection doesn't introduce bias or become gameable,
+			// because `find_path_from_ancestors` uses a HashMap to track the ancestors, which
+			// makes the order in which ancestors are visited non-deterministic.
```

Comment:
I think I was wrong here. Since we're not expecting the runtime to back parachain forks, the nondeterminism added by the HashSet does not matter, because we expect a single path to be able to be formed from the ancestors. Will remove the comment